src/main.c

```diff
@@ -325,7 +327,8 @@ int main(int argc, char** argv)
         };
     }
 
-    if ( (!domain && !domid) || (!address && !setup) || (!setup && ((!json && !sink_list) || !input_path || !input_limit)) )
+    if ( (!domain && !domid) || ((!address == !extended_mark) && !setup) || (!setup && ((!json && !sink_list)
+            || !input_path || (!input_limit == !extended_mark))) )
```

Comment:
I think it's becoming time to split this sanity check here to multiple independent `if`s cause this is now fairly unreadable :)

Reply:
Done.